src/WebJobs.Script/Description/DotNet/PackageManager.cs

```diff
@@ -98,6 +104,139 @@ public static string ResolveNuGetPath()
             return path ?? NuGetFileName;
         }
 
+        public static bool RequiresPackageRestore(string functionPath)
+        {
+            string projectFilePath = Path.Combine(functionPath, DotNetConstants.ProjectFileName);
+            
+            if (!File.Exists(projectFilePath))
+            {
+                // If there's no project.json, we can just return from here 
+                // as there's nothing to restore
+                return false;
+            }
+
+            string lockFilePath = Path.Combine(functionPath, DotNetConstants.ProjectLockFileName);
+
+            if (!File.Exists(lockFilePath))
+            {
+                // If have a project.json and no lock file, we need to 
+                // restore the packages, just return true and skip validation
+                return true;
+            }
+
+            var projectFrameworks = GetProjectDependencies(projectFilePath);
```

Comment:
Also, why do we need to grovel through the files checking things? Can't we simply rely on timestamp or some other indicator that a restore was already done?